spring/jta/src/main/java/org/jboss/narayana/quickstart/spring/xa/DummyXAResource.java

```diff
@@ -0,0 +1,271 @@
+            }
+        };
+        fis.close();
+    }
+
+    public void setFault(faultType fault) {
+        this.fault = fault;
+    }
+
+    public void commit(final Xid xid, final boolean arg1) throws XAException
+    {
+        System.out.println("[" + Thread.currentThread().getName() + "] DummyXAResource (" + serverId + ")      XA_COMMIT  [" + xid + "]");
+        commitRequests += 1;
+
+        if (fault != null) {
+            if (fault.equals(faultType.HALT)) {
+                Runtime.getRuntime().halt(1);
+            }
+        }
+
+        System.out.println("[" + Thread.currentThread().getName() + "] DummyXAResource (" + serverId + ")      commitRequests = " + commitRequests);
+        if (file != null) {
+            if (!file.delete()) {
+                throw new XAException(XAException.XA_RETRY);
+            }
+        }
+        this.xid = null;
+    }
+
+    public boolean isSameRM(XAResource xares) throws XAException {
+        if (xares instanceof DummyXAResource) {
+            DummyXAResource other = (DummyXAResource) xares;
+            if ((this.xid != null && other.xid != null)) {
+                if (this.xid.getFormatId() == other.xid.getFormatId()) {
+                    if (Arrays.equals(this.xid.getGlobalTransactionId(), other.xid.getGlobalTransactionId())) {
+                        if (Arrays.equals(this.xid.getBranchQualifier(), other.xid.getBranchQualifier())) {
+                            return true;
+                        }
+                    }
+                }
+            }
+        }
+        return false;
+    }
+
+
+    public int prepare(final Xid xid) throws XAException
+    {
+        System.out.println("[" + Thread.currentThread().getName() + "] DummyXAResource (" + serverId + ")      XA_PREPARE [" + xid + "]");
+
+        File dir = new File(System.getProperty("user.dir") + "/target/DummyXAResource/" + serverId + "/");
+        dir.mkdirs();
+        file = new File(dir, new Uid().fileStringForm() + "_");
+        try {
+            file.createNewFile();
+            final int formatId = xid.getFormatId();
+            final byte[] gtrid = xid.getGlobalTransactionId();
+            final int gtrid_length = gtrid.length;
+            final byte[] bqual = xid.getBranchQualifier();
+            final int bqual_length = bqual.length;
+
+            DataOutputStream fos = new DataOutputStream(new FileOutputStream(file));
+            fos.writeInt(formatId);
+            fos.writeInt(gtrid_length);
+            fos.write(gtrid, 0, gtrid_length);
+            fos.writeInt(bqual_length);
+            fos.write(bqual, 0, bqual_length);
+            fos.flush();
+            fos.close();
+        } catch (IOException e) {
+            throw new XAException(XAException.XAER_RMERR);
+        }
+        return XA_OK;
+    }
+
+    public Xid[] recover(int flag) throws XAException {
+        Xid toReturn = null;
+        if ((flag & XAResource.TMSTARTRSCAN) == XAResource.TMSTARTRSCAN) {
```

Comment:
I don't think this needs to be a complete implementation, just return the Xid[] with xid or null depending on whether its still available for recovery